platform-sdk/swirlds-virtualmap/src/main/java/com/swirlds/virtualmap/internal/reconnect/PullVirtualTreeResponse.java

```diff
@@ -112,6 +113,11 @@ public void deserialize(final SerializableDataInputStream in, final int version)
         assert learnerView != null;
         path = in.readLong();
         final boolean isClean = in.read() == 0;
+        if (path != Path.ROOT_PATH && learnerView.isLeaf(path)) {
```

Comment:
A single-leaf map has its leaf at the root. Metric will be inconsistent.

Reply:
The reason for the `!= Path.ROOT_PATH` check was to ensure that the `isLeaf()` produces a meaningful result because it relies on the first/last leaf path which is only initialized in the subsequent call to `readNode()`.

However, I see your point. So I removed this extra condition and moved the metric emitting code after the `readNode` call.